### PR TITLE
Suppress minikube upgrade check

### PR DESCRIPTION
**Pull request by meeech**
#481  

Went with `vsminikube` for consistency with `vsdocker` config option

lmk!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -234,6 +234,11 @@
                     "type": "string",
                     "default": null,
                     "description": "Image prefix for docker images ie 'docker.io/brendanburns'"
+                },
+                "vsminikube.checkForUpgrade": {
+                    "type": "boolean",
+                    "default": true,
+                    "description": "Check if update is available for minikube?"
```
Comment (itowlson) [inline package.json]: Suggest minor wording tweak:
```suggestion
                    "description": "Notify on startup if update is available for minikube"
```

File: src/components/clusterprovider/minikube/minikube.ts [hunk 2]
```diff
@@ -84,7 +84,10 @@ class MinikubeImpl implements Minikube {
 }
 
 async function minikubeUpgradeAvailable(context: Context): Promise<void> {
-    if (!await checkPresent(context, CheckPresentMode.Silent)) {
+
+    const checkForUpgrade = vscode.workspace.getConfiguration().get("vsminikube.checkForUpgrade");
```
Comment (itowlson) [inline src/components/clusterprovider/minikube/minikube.ts]: I've been trying to contain configuration checks and keys within the `components/config/config.ts` file - would you mind encapsulating this there?

File: src/components/clusterprovider/minikube/minikube.ts [hunk 3]
```diff
@@ -84,7 +84,10 @@ class MinikubeImpl implements Minikube {
...
+
+    if (!await checkPresent(context, CheckPresentMode.Silent) || !checkForUpgrade) {
```
Comment (itowlson) [inline src/components/clusterprovider/minikube/minikube.ts]: Either amend the comment on the next line, or break this out into two statements:

```
if (!await ...) {
  // not installed...
  return;
}
if (!checkForUpgrade) {
  // don't bother the user
  return;
}
```

Perhaps even better introduce an explaining variable:

```
const performUpgradeCheck = await checkPresent(...) && checkForUpgrade;
if (!performUpgradeCheck) {
  return;
}
```

File: README.md [hunk 4]
```diff
@@ -63,6 +63,14 @@ for your cluster to run it. To do this, add the following to your VS Code prefer
 
 where `<your-image-prefix-here>` is something like `docker.io/brendanburns`.
 
+```javascript
```
[the remark this answers is not in the document]
Comment (meeech) [reply README.md]: 👍

**Comment by meeech** [comment]
@itowlson I realized after I just updated the pr I might have mis-understood where you wanted the config flag. Did you mean that I should put it at the same level as `disable-lint` ? lmk

**Comment by itowlson** [comment]
@meeech Yes, same level as `disable-lint` is what we should aim for.  Sorry for the confusing explanation!

**Comment by meeech** [comment]
@iotgwbot 👍

**Comment by itowlson** [comment]
There's a couple of cosmetic lint issues causing CI to fail (a missing semicolon and consecutive blank lines).  Are you okay to fix these?  If you don't have time then I can do it.  Thanks!

**Comment by meeech** [comment]
all done. still lint issues, but those seem due to some warning, not the code itself.

**Comment by itowlson** [comment]
Yeah, the lint warnings were about cosmetic coding standards, purely style stuff.  I made the style tweaks and merged the PR.  Thanks again!

**Comment by meeech** [comment]
Ok thanks. Sorry about that. I thought I pushed up those 2 fixes (semi colon and extra space) and was getting some other warnings. I must have messed something up. :/

**Pull request merged**